### PR TITLE
Added node template support for hetzner cloud

### PR DESCRIPTION
**Pull request by ntoljic**
This PR is the basis for solving #105. Rather than solving the problem for all providers, this PR adds specific support for the Hetzner Cloud driver.

To make this work, the rancher instance must have the Hetzner Cloud driver installed. The following terraform snippet, shows how this can be achieved:

```terraform
# Add the Hetzner node driver
resource "rancher2_node_driver" "hetzner" {
    active = true
    builtin = false
    description = "Rancher UI driver for the Hetzner Cloud docker driver."
    external_id = "https://github.com/mxschmitt/ui-driver-hetzner"
    name = "hetzner"
    ui_url = "https://storage.googleapis.com/hcloud-rancher-v2-ui-driver/component.js"
    url = "https://github.com/JonasProgrammer/docker-machine-driver-hetzner/releases/download/2.0.1/docker-machine-driver-hetzner_2.0.1_linux_amd64.tar.gz"
    whitelist_domains = ["storage.googleapis.com"]
}
```

The `external_id` contains the [link](https://github.com/mxschmitt/ui-driver-hetzner) to the Github repo for the UI. From what I could find, the repo uses a custom [docker machine driver](https://jonasprogrammer.github.io/docker-machine-driver-hetzner/) to do the actual work of setting up the Hetzner VMs.

The following snippet shows how the config can be used to create a node template:

```terraform 
resource "rancher2_node_template" "my_hetzner_node_template" {
  depends_on = [rancher2_node_driver.hetzner]
  engine_install_url = "https://get.docker.com"
  name = "my_hetzner_node_template"
  hetzner_config {
    api_token = <Hetzner Token goes here>
    image = "ubuntu-18.04"
    server_location = "nbg1"
    server_type = "cx11"
  }
}
```

## Caveats

I have not quite figured out how to write the tests for this. The existing examples all use built-in drivers with cloud credentials. Would be happy to finish this, but would need some pointers.

### CODE REVIEW

File: rancher2/schema_node_template_hetznercloud.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
+package rancher2
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
```
Comment (rawmind0) [inline rancher2/schema_node_template_hetznercloud.go]: This import is not correct, please add correct schema import to build correctly

```
-	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
```

**Review by rawmind0 — CHANGES_REQUESTED** [review]

File: rancher2/schema_node_template_hetznercloud.go (hunk 1 above)
Comment (ntoljic) [reply rancher2/schema_node_template_hetznercloud.go]: @rawmind0, thanks for pointing this out. Fixed it and the CI build is now passing. Keep in mind though that I have not added any tests.

Comment (ntoljic) [reply rancher2/schema_node_template_hetznercloud.go]: @rawmind0, is there anything else you need from me to merge this change? I must admit that I'm fairly new to the PR process on github.

File: go.mod [hunk 2]
```diff
@@ -3,8 +3,8 @@ module github.com/terraform-providers/terraform-provider-rancher2
 go 1.12
 
 require (
-	github.com/google/btree v1.0.0 // indirect
 	github.com/hashicorp/go-version v1.2.0
+	github.com/hashicorp/terraform v0.12.19
```
Comment (rawmind0) [inline go.mod:7]: @ntoljic , why do you need these changes on `go.mod`?? Provider is building fine without them.

**Review by rawmind0 — CHANGES_REQUESTED** [review]
@ntoljic , this node template is not working due to the id for user created node_driver is missed. Have you tested it??

Following your instructions, getting error `Error: [ERROR] Getting Node Driver hetzner: Bad response statusCode [404]. Status [404 Not Found]. Body: [baseType=error, code=NotFound, message=nodedrivers.management.cattle.io "hetzner" not found] from [https://XX.XX.XX.XX/v3/nodedrivers/hetzner]`

I've submitted a PR to address this issue, adding `driver_id` argument to `rancher2_node_template` resource, https://github.com/terraform-providers/terraform-provider-rancher2/pull/240 . Once it's merged, could you please rebase and test your PR with this tf file??

```
resource "rancher2_node_driver" "hetzner" {
    active = true
    builtin = false
    description = "Rancher UI driver for the Hetzner Cloud docker driver."
    external_id = "https://github.com/mxschmitt/ui-driver-hetzner"
    name = "hetzner"
    ui_url = "https://storage.googleapis.com/hcloud-rancher-v2-ui-driver/component.js"
    url = "https://github.com/JonasProgrammer/docker-machine-driver-hetzner/releases/download/2.0.1/docker-machine-driver-hetzner_2.0.1_linux_amd64.tar.gz"
    whitelist_domains = ["storage.googleapis.com"]
}

resource "rancher2_node_template" "my_hetzner_node_template" {
  engine_install_url = "https://get.docker.com"
  name = "my_hetzner_node_template"
  driver_id = rancher2_node_driver.hetzner.id
  hetzner_config {
    api_token = "XXXXXXXXXX"
    image = "ubuntu-18.04"
    server_location = "nbg1"
    server_type = "cx11"
  }
}
```

Also, node_template documentation and tests should be updated  and included in the PR. For adding test, please take a look on `rancher2/resource_rancher2_node_template_test.go` and add another case for using your node_driver. You shouldn't  provide token or any sensible data, take a look to other node_template tests.

**Comment by rawmind0** [comment]
This PR requires #240

**Comment by ntoljic** [comment]
@rawmind0, I must have checked in an outdated version. I recently switched machines and must have forgot to commit/push the latest changes.
As it stands this PR is useless and unfortunately I currently don't have the time to work on it.

**Pull request closed without merging**